### PR TITLE
Refactor shell scripts and add Docker-based integration tests

### PR DESCRIPTION
**Pull request by zachallaun**
* Add `bin/activate_version_manager.sh`

This script encapsulates detecting, finding, and activating a version manager. Currently asdf and rtx are supported, though more work likely needs to be done to make sure that they can be found/activated in more cases.

* Refactor `bin/start_lexical.sh`

This script now uses `activate_version_manager.sh` to activate the version manager. It additionally ensures that it exits with an error code if any commands run exit non-zero.

* `bin/boot.exs` halts after booting Lexical if `LX_HALT_AFTER_BOOT` is set

* Begin using `shellcheck` and `shfmt` for script linting and formatting

Both of these tools can be installed using asdf/rtx and have editor support through plugins. CI is not currently enforcing that these be used; we can decide on that later.

* Add `integration/` dir and integration tests for booting Lexical

Currently, these test system installations of elixir, asdf, and rtx, all using bash and running in a Docker container. I'd like to add tests using different shells (zsh/fish/etc.) at some point as well.

These tests are not currently running in CI. That will be added in a later commit.

### CODE REVIEW

File: integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+COPY integration/boot/set_up_rtx.sh integration/boot/set_up_rtx.sh
+RUN integration/boot/set_up_rtx.sh
+
+COPY integration/boot/set_up_asdf.sh integration/boot/set_up_asdf.sh
+RUN integration/boot/set_up_asdf.sh
+
+COPY apps apps
+COPY bin bin
+COPY config config
+COPY priv priv
```
Comment (Blond11516) [inline integration/Dockerfile:33]:
```suggestion
```
I had to remove this to make build work. I don't have a priv directory at the project root.

File: integration/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  openssl \
+  libssl-dev
+
+WORKDIR /lexical
+
+COPY integration/boot/set_up_rtx.sh integration/boot/set_up_rtx.sh
+RUN integration/boot/set_up_rtx.sh
+
+COPY integration/boot/set_up_asdf.sh integration/boot/set_up_asdf.sh
+RUN integration/boot/set_up_asdf.sh
```
Comment (Blond11516) [inline integration/Dockerfile:28]: Installing Erlang twice in a row is rough 🙃 It's fine for local work since it's cached, but I think we'll probably want to push the image to Github Container Registry if/when we get to putting this in CI.

File: integration/test.sh [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    if [[ $1 != "" ]]; then
+        command="$1 && $command"
+    fi
+
+    docker run -i lx bash -c "$command" 2>&1
+    return $?
+}
+
+run_test() {
```
Comment (Blond11516) [inline integration/test.sh:25]: Should this go in `test_utils.sh`?

**Review by Blond11516 — COMMENTED** [review]
Tested locally, works great!

This is really good stuff, it will make working on the boot process much easier (and less scary).

File: integration/Dockerfile (hunk 1 above)
Comment (zachallaun) [reply integration/Dockerfile:33]: Thanks for pointing this out! I don't think the root `priv` ever actually needs to be copied. It only contains cached PLTs from running Dialyzer.

File: integration/test.sh (hunk 3 above)
Comment (zachallaun) [reply integration/test.sh:25]: As it's currently written, I don't think so, because it calls `start_lexical` directly. I originally had a more abstracted version of it in `test_utils.sh`, but ended up moving it here because there was just a lot of nested output/exit code capture going on and I found it hard to follow/debug.

My inclination is to leave it here until some of this logic is needed elsewhere, then pull it out.

Comment (Blond11516) [reply integration/test.sh:25]: Sounds like a good plan 👍

File: apps/remote_control/priv/port_wrapper.sh [hunk 4]
```diff
@@ -1,42 +1,21 @@
...
-# Clean up
-wait $pid1
-ret=$?
-kill -KILL $pid2
-exit $ret
+# Clean up and exit with wrapped program's exit code
+wait $wrapped_pid
+wrapped_ret=$?
+kill -KILL $monitor_pid
+exit $wrapped_ret
```
[the remark this answers is not in the document]
Comment (zachallaun) [reply apps/remote_control/priv/port_wrapper.sh:21]: Yep, this was just my attempt to rewrite/reword things in a way that I thought would make more sense for someone coming to this for the first time.

**Comment by zachallaun** [comment]
This branch fixes a bug that's currently on main that prevents certain environments using `rtx` from starting the remote node.

File: apps/remote_control/priv/port_wrapper.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (scohen) [reply apps/remote_control/priv/port_wrapper.sh:21]: Just link to the docs

**Review by scohen — APPROVED** [review]
